opentracing-impl/src/main/java/io/opentracing/concurrent/TracedExecutorService.java

```diff
@@ -0,0 +1,102 @@
+package io.opentracing.concurrent;
+
+import io.opentracing.ActiveSpanManager;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.impl.GlobalTracer;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+public class TracedExecutorService implements ExecutorService {
+    private ExecutorService executor;
+    private ActiveSpanManager manager;
```

Comment:
note how this thing does *not* need a Tracer ref anymore, and it's also pretty easy to write given an ASM.